src/write.rs

```diff
@@ -222,15 +222,26 @@ where
                         .unwrap()
                 };
 
+                let free_elements = w_handle.data.capacity() - w_handle.data.len();
+
+                if free_elements < r_handle.data.len() {
+                    // pre-allocate if possible
+                    w_handle.data.reserve(r_handle.data.len() - free_elements);
+                }
```

Comment:
I believe (from the comment above) that `w_handle.data` should be empty here, which means that we can probably do better than this. Specifically, I think below we can replace
```rust
w_handle.data.extend(r_handle.data.iter_mut().map(...))
```
with
```rust
mem::replace(&mut w_handle.data, r_handle.data.iter_mut().map(...).collect())
```
It'd be good to assert that the returned `.data` is indeed empty there though, and then run the test suite.

Reply:
An empty `std` collection always consumes no memory in Rust, so I don't think that's true?

Reply:
Also, I'm pretty sure `extend` also uses the pre-allocate on `ExactSizeIterator` trick, so this may all be unimportant.

Reply:
Only for a fresh new one, but one that previously had elements but they were all removed still keeps that allocated memory around. It would be more accurate to say they are lazily allocated.

Reply:
True, but in this case, unless I'm misremembering, they should truly be empty (as in, freshly allocated). See the comment above in the code.

Reply:
Ah, I see what you mean. Still, it doesn't really matter. `collect()` will simply call `extend` internally.

Reply:
Yup, which in turn uses the `ExactSizeIterator` to pre-reserve when possible.